.github/workflows/fast_testing.yaml

```diff
@@ -39,10 +40,49 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Install tarantool ${{ matrix.tarantool }}
+        if: matrix.tarantool != 'debug'
         uses: tarantool/setup-tarantool@v1
         with:
           tarantool-version: ${{ matrix.tarantool }}
 
+      - name: Get last commit from Tarantool
+        if: matrix.tarantool == 'debug'
+        id: get-tnt-commit
+        run: |
+          commit_hash=$(git ls-remote https://github.com/tarantool/tarantool.git HEAD | head -c 8)
+          echo "::set-output name=hash::${commit_hash}"
+        shell: bash
+
+      - name: Cache tarantool build
+        if: matrix.tarantool == 'debug'
+        id: cache-tnt-debug
+        uses: actions/cache@v3
+        with:
+          path: ~/tnt-build/
+          key: cache-tnt-${{ matrix.tarantool }}-${{ steps.get-tnt-commit.outputs.hash }}
+
+      - name: Build tarantool ${{ matrix.tarantool }}
+        if: matrix.tarantool == 'debug' && steps.cache-tnt-debug.outputs.cache-hit != 'true'
+        run: |
+          sudo apt-get -y install git build-essential cmake make zlib1g-dev \
+            libreadline-dev libncurses5-dev libssl-dev \
+            libunwind-dev libicu-dev python3 python3-yaml \
+            python3-six python3-gevent
+          cd ~/
+          git clone https://github.com/tarantool/tarantool.git --recursive
+          cd tarantool
+          git submodule update --init --recursive
+          mkdir build && cd build
+          cmake .. -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=~/tnt-build/
+          make
+          sudo make install
+          sudo cp extra/dist/tarantoolctl ~/tnt-build/bin/
+
+      - name: Install tarantool ${{ matrix.tarantool }}
+        if: matrix.tarantool == 'debug'
+        run: |
+          sudo cp -rvaP ~/tnt-build/* /usr/local/
```

Comment:
The problem is in actions/cache. It doesn't have permissions to restore data in  `/usr/local/`:

```
Run actions/cache@v3
  with:
    path: /usr/local/bin/tarantoolctl
  /usr/local/bin/tarantool
  /usr/local/include/tarantool/
  /usr/local/lib/tarantool
  /usr/local/share/tarantool/
  
    key: cache-tnt-debug-master-3dd483d1
  env:
    version_postfix: -3dd483d1
Received 6132740 of 10327044 (59.4%), 5.8 MBs/sec
Received 10327044 of 10327044 (100.0%), 9.3 MBs/sec
Cache Size: ~10 MB (10327044 B)
/usr/bin/tar --use-compress-program zstd -d -xf /home/runner/work/_temp/f63d7112-aed2-4263-95e7-d3fca9a9d1b5/cache.tzst -P -C /home/runner/work/expirationd/expirationd
/usr/bin/tar: ../../../../../usr/local/include/tarantool: Cannot mkdir: Permission denied
/usr/bin/tar: ../../../../../usr/local/include/tarantool: Cannot mkdir: Permission denied
/usr/bin/tar: ../../../../../usr/local/include/tarantool/luajit.h: Cannot open: No such file or directory
/usr/bin/tar: ../../../../../usr/local/include/tarantool: Cannot mkdir: Permission denied
/usr/bin/tar: ../../../../../usr/local/include/tarantool/luaconf.h: Cannot open: No such file or directory
/usr/bin/tar: ../../../../../usr/local/include/tarantool: Cannot mkdir: Permission denied
/usr/bin/tar: ../../../../../usr/local/include/tarantool/lua.h: Cannot open: No such file or di
```

As an alternative, it is possible to add ~/tnt-debug/bin/ to PATH and maybe it will work. But I think it's a matter of taste.